app/autodiscovery/providersxml/src/main/java/com/fsck/k9/autodiscovery/providersxml/ProvidersXmlDiscovery.kt

```diff
@@ -135,7 +135,7 @@ class ProvidersXmlDiscovery(
         }
 
         // TODO: Remove this hack
-        val authType = if (host == "smtp.gmail.com" || host == "smtp.googlemail.com") {
+        val authType = if (host == "smtp.gmail.com" || host == "smtp.googlemail.com" || host == "smtp-mail.outlook.com" || host == "smtp.office365.com") {
```

Comment:
How about using `findOAuthConfiguration` here and checking for null (maybe move the method to a new class)? That would avoid having to specify the URLs in multiple different places.

Reply:
Yes, this is discussed in https://github.com/k9mail/k-9/pull/6094#discussion_r888016849 too.
@cketti : ok about reusing `findOAuthConfiguration` from here?

Reply:
I created PR #6097 to clean this up.

Reply:
Thanks, it's indeed much simpler.